### PR TITLE
feat: deprecate BaseBToken in favour of OZ ERC20

### PR DESCRIPTION
**Pull request by 0xteddybear**
closes BAL-77

feedback wanted on

- [x] should I be concerned on not getting any test fails since I changed the semantics of the BToken? (error messages are different at least, and I didn't have to modify any tests)
- [x] should I add tests for the {increase,decrease}Allowance methods? try to instead use an external implementation that already provides them?

### CODE REVIEW

File: .solhint.json
Comment (wei3erHase) [inline .solhint.json:1]: why in this PR? 😢

**Review by wei3erHase — COMMENTED** [review]
logic seems solid, blz clean up PR 🙏

File: .solhint.json
Comment (0xteddybear) [reply .solhint.json:1]: it's on top of the previous branch: #65. I'll tear them apart.

File: src/contracts/BPool.sol [hunk 1]
```diff
@@ -332,7 +335,7 @@ contract BPool is BBronze, BToken, BMath, IBPool {
     require(tokenAmountOut <= bmul(tokenOutBalance, MAX_OUT_RATIO), 'ERR_MAX_OUT_RATIO');
 
     poolAmountIn =
-      calcPoolInGivenSingleOut(tokenOutBalance, outRecord.denorm, _totalSupply, _totalWeight, tokenAmountOut, _swapFee);
+      calcPoolInGivenSingleOut(tokenOutBalance, outRecord.denorm, totalSupply(), _totalWeight, tokenAmountOut, _swapFee);
```
Comment (wei3erHase) [inline src/contracts/BPool.sol:338]: yikes, should we increase the linter line length to 125?

**Review by wei3erHase — COMMENTED** [review]
- as for the tests, i don't think we were testing the BToken yet, given is an OZ library, don't think we should be testing it either. so no worries bc of no tests failing
- as for testing increase and decrease allowance, since we're maintaining that feature, we should at least unit test it imo, and perhaps add for the inheritable internals `_push` and `_pull`, so that if someone later inherits BToken has all the features tested

**Review by wei3erHase — APPROVED** [review]
looks good! 👏

**Pull request merged**